docs/source/starthere/fundamentals.rst

```diff
@@ -0,0 +1,240 @@
+
+
+Example training script
+-----------------------
+
+Putting it all together, here is an example training script for our ``ExampleEncDecModel`` model. We highlight the 3 most important lines, which put together everything we discussed in the previous section.
+
+.. code-block:: python
+	:caption: run_example_training.py
+	:linenos:
+	:emphasize-lines: 10, 11, 12
+
+	import pytorch_lightning as pl
+	from nemo.collections.path_to_model_class import ExampleEncDecModel
+	from nemo.core.config import hydra_runner
+
+	@hydra_runner(
+		config_path="config_file_dir_path",
+		config_name="config_file_name"
+	)
+	def main(cfg):
+		trainer = pl.Trainer(**cfg.trainer)
+		model = ExampleEncDecModel(cfg.model, trainer)
+		trainer.fit(model)
+
+	if __name__ == '__main__':
+		main(cfg)
+
+
+Let's go through the code:
+
+* *Lines 1-3*: import statements (second one is made up for the example).
+* *Lines 5-8*: a decorator on lines 5-8 of ``run_example_training.py`` will look for a config file at ``{config_path}/{config_name}.yaml``, and load its contents into the ``cfg`` object that is passed into the ``main`` function. This functionality is provided by `Hydra <https://hydra.cc/docs/intro/>`__. Instead of a YAML file, we could also have specified the default config as a dataclass, and passed that into the ``@hydra_runner`` decorator.
+* *Line 7*: initialize a PTL trainer object, using the parameters specified in the ``trainer`` section of the config.
+* *Line 8*: initialize a NeMo model, passing in both the parameters in the ``model`` section of the config, and a PTL trainer.
+* *Line 9*: call ``trainer.fit`` on the model. This one unassuming line will carry out our entire training process. PTL will make sure we iterate over our data and call the ``training_step`` we define for each batch (as well as any other PTL `callbacks <https://lightning.ai/docs/pytorch/stable/extensions/callbacks.html>`__ that may have been defined).
+
+
+
+Overriding configs
+------------------
+
+The ``cfg`` object in the script above is a dictionary-like object that contains our configuration parameters. Specifically, it is an `OmegaConf <https://omegaconf.readthedocs.io/en/2.3_branch/usage.html>`__ ``DictConfig`` object. These objects have special features such as dot-notation `access <https://omegaconf.readthedocs.io/en/latest/usage.html#access>`__, `variable interpolation <https://omegaconf.readthedocs.io/en/latest/usage.html#variable-interpolation>`__ and ability to set `mandatory values <https://omegaconf.readthedocs.io/en/latest/usage.html#mandatory-values>`__.
+
+We can run the script above like this:
+
+.. code-block:: bash
+
+	python run_example_training.py
+
+This will use the default config file specified inside the ``@hydra_runner`` decorator.
```

Comment:
revise.

The script will use the default config file specified inside the ``@hydra_runner`` decorator.